### PR TITLE
fix: separate android beta distributions to fix memory issues in CI

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **fix**

This PR resolves [CX-1966]

### Description

Co-authored with @artsyjian 

Separates our one android beta deployment job into 2 jobs, 1 that distributes to google play and one that distributes to firebase in order to workaround memory issues we were seeing in CI.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Separate android beta deploys into 2 jobs - Brian, Jian

</details>

[CX-1966]: https://artsyproduct.atlassian.net/browse/CX-1966

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 60524c49fb78f5cd22039c60439606da065786c4
</p>

**Review by artsyjian — APPROVED** [review]
lgtm

**Comment by brainbicycle** [comment]
Going to merge so we can hopefully have some up to date betas for tomorrow! Let me know if any concerns

File: scripts/deploy-if-beta-branch-android [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 
-
 source ./scripts/source-for-bash-env
 
 if [ "${LOCAL_BRANCH}" == "beta-android" ]; then
-	./scripts/distribute-android
-fi
+  DISTRIBUTION_TARGET="${1-"PLAY_STORE"}"
+  ./scripts/distribute-android $DISTRIBUTION_TARGET
+fi
```
Comment (pvinis) [inline scripts/deploy-if-beta-branch-android:9]:
```suggestion
fi

```

File: scripts/distribute-android [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
-
 ./scripts/setup-distribute-linux
 
-bundle exec fastlane ship_beta_android
+DISTRIBUTION_TARGET="${1-"PLAY_STORE"}"
+if [ $DISTRIBUTION_TARGET == "FIREBASE" ]; then
+  bundle exec fastlane ship_beta_android_firebase
+elif [ $DISTRIBUTION_TARGET == "PLAY_STORE" ]; then
+  bundle exec fastlane ship_beta_android_play_store
+fi
```
Comment (pvinis) [inline scripts/distribute-android:11]:
```suggestion
fi

```

**Comment by brainbicycle** [comment]
that status check that is pending is now removed, going to merge and then update our required status checks

**Pull request merged**